### PR TITLE
[CCI] Create document lifecycle guide

### LINKED ISSUE #352: [Guide] Document Lifecycle
Create `guides/document_lifecycle.md` similar to the [Ruby guide](https://github.com/opensearch-project/opensearch-ruby/blob/main/guides/document_lifecycle.md)

You must assure that: 
- All code-blocks, when run from start to finish in order they appear in the guide, do not throw any errors. If you want to demonstrate that certain statements will throw errors, they must be wrapped in a `try/catch` and print out the expected error.
- All code-blocks are properly marked with the language it's written in (i.e.  ` ```ruby `)
- The instructions, though mostly identical to the Ruby guide, are suitable for this repo.

### PR DESCRIPTION
**Pull request by Nicksqain**
### Description

- [x] Create document lifecycle guide

### Issues Resolved

- resolves #352 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by saimedhi** [comment]
Hello @Nicksqain, please merge all the commits into a single commit. Thanks.

File: guides/document_lifecycle.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+client.index(index=index, id=2, body={'title': 'The Lion King', 'year': 1994})
+```
+
+### Create a new document with auto-generated ID
+
+You can also create a new document with an auto-generated ID by omitting the `id` parameter. The following code creates documents with an auto-generated IDs in the `movies` index:
+
+```python
+print(client.create(index=index, body={'title': 'The Lion King 2', 'year': 1998}))
+# OR client.index(index=index, body={"title": "The Lion King 2", "year": 1998})
```
Comment (saimedhi) [inline guides/document_lifecycle.md:56]: The commented code worked.

**Review by saimedhi — CHANGES_REQUESTED** [review]

**Comment by Nicksqain** [comment]
@dblock Yes, I'll be sure to add the necessary changes on a free day!

**Comment by dblock** [comment]
Superseded with https://github.com/opensearch-project/opensearch-py/pull/559.

**Pull request closed without merging**